src/test/java/online/bookstore/book/BookControllerTest.java

```diff
@@ -0,0 +1,221 @@
+        books.add(new BookDto()
+                .setId(3L)
+                .setTitle("Book Three")
+                .setAuthor("Author Three")
+                .setPrice(BigDecimal.valueOf(39.99))
+                .setIsbn("978-3-16-148412-4")
+                .setDescription("Description for book three")
+                .setCoverImage("http://example.com/book3.jpg")
+                .setCategories(Set.of()));
+        Pageable pageable = PageRequest.of(0, 3);
+        Page<BookDto> bookPage = new PageImpl<>(books, pageable, books.size());
+        when(bookService.getAll(pageable)).thenReturn(bookPage);
+        MvcResult result = mockMvc.perform(get("/books")
+                        .param("page", "0")
+                        .param("size", "3")
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andReturn();
+        String jsonResponse = result.getResponse().getContentAsString();
+        List<BookDto> actualBooks = objectMapper.readValue(
+                objectMapper.readTree(jsonResponse).get("content").toString(),
+                new TypeReference<List<BookDto>>() {}
+        );
+
+        for (int i = 0; i < books.size(); i++) {
+            BookDto expectedBook = books.get(i);
+            BookDto actualBook = actualBooks.get(i);
+            EqualsBuilder.reflectionEquals(expectedBook, actualBook, "categories");
+        }
+    }
+
+    @Test
+    @WithMockUser(username = "user", roles = {"USER"})
+    void getById_WithValidId_ShouldReturnBookDto() throws Exception {
+        Long bookId = 1L;
+        BookDto expected = new BookDto()
+                .setId(bookId)
+                .setTitle("Book One")
+                .setAuthor("Author One")
+                .setPrice(BigDecimal.valueOf(19.99))
+                .setIsbn("978-3-16-148410-0")
+                .setDescription("Description for book one")
+                .setCoverImage("http://example.com/book1.jpg")
+                .setCategories(Set.of());
+        when(bookService.getById(bookId)).thenReturn(expected);
+        MvcResult result = mockMvc.perform(get("/books/{id}", bookId)
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andReturn();
+        String jsonResponse = result.getResponse().getContentAsString();
+        BookDto actual = objectMapper.readValue(jsonResponse, BookDto.class);
+        EqualsBuilder.reflectionEquals(expected, actual, "categories");
+    }
+
+    @Test
+    @WithMockUser(username = "admin", roles = {"ADMIN"})
+    @Sql(
+            scripts = "classpath:database/books/delete-new-book.sql",
+            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD
+    )
+    @Sql(
+            scripts = "classpath:database/books/delete-fiction-category.sql",
+            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD
+    )
+    @Transactional
```

Comment:
if your test doesn’t work without this annotation `@SqlMergeMode(SqlMergeMode.MergeMode.MERGE)` can help you